src/Components/Components/src/Rendering/RenderTreeBuilder.cs

```diff
@@ -663,6 +680,39 @@ public void AddComponentReferenceCapture(int sequence, Action<object> componentR
         _lastNonAttributeFrameType = RenderTreeFrameType.ComponentReferenceCapture;
     }
 
+    /// <summary>
+    /// Adds a frame indicating the render mode on the enclosing component frame.
+    /// </summary>
+    /// <param name="sequence">An integer that represents the position of the instruction in the source code.</param>
+    /// <param name="renderMode">The <see cref="IComponentRenderMode"/>.</param>
+    public void AddComponentRenderMode(int sequence, IComponentRenderMode renderMode)
```

Comment:
This is the new API that the Razor compiler should call when it sees `@rendermode=something` on a component call site (e.g., `<SomeComponent @rendermode="Server" />`).

Reply:
Removed